### PR TITLE
Wisdom stat affects health restored through potions

### LINKED ISSUE #57: Implement WIS - Wisdom
**User Story**
As a game designer, I'd like a player's wisdom score/modifier to affect the health restored by health potions.

**Acceptance Criteria**

- The health restored by potions scales with the wisdom ability modifier

**Notes**

---

**Why is this feature needed? Please describe the problem your requested feature wants to solve**
Wisdom does not currently affect any elements of gameplay, but it should.

**Describe the solution you'd like**
The health restored by health potions should increase with a character's wisdom ability modifier (which itself is affected by the wisdom ability score). The following calculation is proposed:

`base_amount + (ability_bonus > 0) ? 2 ^ ability_bonus:0`

**Use Case**
Health potions restore 15 health. Player A has 14 WIS (modifier of +2). Player B has 10 WIS (modifier of +0). Player C has 6 WIS (modifier of -2).

A: `2 > 0 === true, so 15 + 2 ^ 2 = 15 + 4 = 19`
B: `0 > 0 === false, so 15 + 0 = 15`
C: `-2 > 0 === false, so 15 + 0 = 15`

### PR DESCRIPTION
**Pull request by Frosty273**
GitHub Issue (If applicable): #57 

## PR Type

What kind of change does this PR introduce?

## What is the current behavior?

The player can currently invest stat points into the wisdom stat which currently doesn't affect anything.

## What is the new behavior?

The wisdom stat is used to modify the amount of health restored through potions used by the player.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [ ] Added new tests to the test suite for the feature
- [ ] Tested the full test suite, and ensured that the feature does not break current build.
- [ ] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [ ] Checked licensing of Frameworks/Libraries (if applicable) 

## Other information

Internal Issue (If applicable):

Closes #57

### CODE REVIEW

File: src/components/view-item/index.js [hunk 1]
```diff
@@ -54,9 +56,15 @@ const ViewItem = ({
...
         case 'potion':
+            data.hp = data.hpReset;
+            let potionRestore = calculateWisdomPotionBonus(
+                data.hp,
+                calculateModifier(stats.abilities.wisdom)
+            );
+            data.hp = potionRestore;
             itemStats.push(
                 <StatsItem
-                    stats={{ name: 'heal', value: data.hp }}
```
Comment (Frosty273) [inline src/components/view-item/index.js:59]: On a second look through, "data.hp" didn't have to be changed to "potionRestore" as they would have done the same thing. Can change back if required.

Comment (Pyxxil) [reply src/components/view-item/index.js:59]: Would that also remove the need for the hpReset?

Comment (Frosty273) [reply src/components/view-item/index.js:59]: So the purpose of hpReset was that closing and reopening of a potion object continuously recalls the function, so the amount to be restored continues to increase when it should only be static and run once. Therefore, I implemented the hpReset attribute to reset to its initial value. It was also simpler to alter the data.hp value for the amount to be restored.

Comment (Pyxxil) [reply src/components/view-item/index.js:59]: Yeah, but, if you remove the 'data.hp = hpRestore' line, then wouldn't that keep the value consistent? Or am I misunderstanding?

Comment (Frosty273) [reply src/components/view-item/index.js:59]: When using a potion, the health restored is taken from data.hp, so the wisdom bonus will need to adjust the value of data.hp to account for the new value. Without that line, only the value displayed to increase would be correct and the actual amount restored would still be the default amount (15 in the case of a regular potion).

Comment (Pyxxil) [reply src/components/view-item/index.js:59]: Ohh, okay. Then yeah, it's up to you if you want to change what you said. I don't think it makes much difference.

Comment (Pyxxil) [reply src/components/view-item/index.js:59]: Ohh, okay. Then yeah, it's up to you if you want to change what you said. I don't think it makes much difference.

Comment (Frosty273) [reply src/components/view-item/index.js:59]: It doesn't need to be changed imo as it does the same thing but if it irritates some people then I can change it.

**Review by Pyxxil — APPROVED** [review]
I don't think changing that line should stop this, but perhaps we'll need to rethink the amount this restores at some point (but that's a seperate issue).

Looks good!

**Pull request merged**